src/test/store/transforms.test.js

```diff
@@ -152,3 +155,228 @@ describe('focus subtree transform', function() {
     });
   });
 });
+
+describe('"merge-call-node" transform', function() {
+  describe('on a call tree', function() {
+    /**
+     * Assert this transformation:
+     *
+     *                     A:3,0                              A:3,0
+     *                       |                                  |
+     *                       v       Focus [A, B, C]            v
+     *                     B:3,0           -->                B:3,0
+     *                     /    \                           /   |   \
+     *                    v      v                         v    v    v
+     *                C:2,0     H:1,0                 D:1,0   F:1,0   I:1,1
+     *               /      \         \                 |       |
+     *              v        v         v                v       v
+     *            D:1,0     F:1,0     I:1,1           E:1,1   G:1,1
+     *            |           |
+     *            v           v
+     *          E:1,1       G:1,1
+     */
+    const profile = getProfileForUnfilteredCallTree();
+    const { dispatch, getState } = storeWithProfile(profile);
+    const originalCallTree = selectedThreadSelectors.getCallTree(getState());
+    const threadIndex = 0;
+    const A = 0;
+    const B = 1;
+    const C = 2;
+
+    it('starts as an unfiltered call tree', function() {
+      expect(formatTree(originalCallTree)).toMatchSnapshot();
+    });
+
+    it('call node [A, B, C] can be merged into [A, B]', function() {
+      dispatch(
+        addTransformToStack(threadIndex, {
+          type: 'merge-call-node',
+          callNodePath: [A, B, C],
+          implementation: 'combined',
+          inverted: false,
+        })
+      );
+      const callTree = selectedThreadSelectors.getCallTree(getState());
+      expect(formatTree(callTree)).toMatchSnapshot();
+    });
+  });
+
+  describe('on a JS call tree', function() {
+    const profile = getProfileWithMixedJSImplementation();
+    const threadIndex = 0;
+
+    // funcIndexes in the profile fixture.
+    /* eslint-disable no-unused-vars */
+    const RUN_SCRIPT = 0;
+    const ON_LOAD = 1;
+    const A = 2;
+    const B = 3;
+    const ION_CANNON = 4;
+    /* eslint-enable no-unused-vars */
+
+    const mergeJSPathAB = {
+      type: 'merge-call-node',
+      callNodePath: [ON_LOAD, A],
+      implementation: 'js',
+      inverted: false,
+    };
+
+    const mergeCombinedPathToA = {
+      type: 'merge-call-node',
+      callNodePath: [RUN_SCRIPT, ON_LOAD, A],
+      implementation: 'combined',
+      inverted: false,
+    };
+
+    it('starts as an untransformed call tree', function() {
+      /**
+       *     JS::RunScript
+       *          ↓
+       *        onLoad
+       *      ↙       ↘
+       *     a        js::jit::IonCannon
+       *     ↓             ↓
+       *     b             a
+       *                   ↓
+       *                   b
+       */
```

Comment:
Thanks for putting all these trees into these tests. Makes it much easier to read.